From c49dd772f915336b2c1b469a3c173f0a3ee434e4 Mon Sep 17 00:00:00 2001
From: Stefano Babic <sbabic@denx.de>
Date: Fri, 12 Oct 2018 12:00:50 +0200
Subject: [PATCH] Fix config for OE

Signed-off-by: Stefano Babic <sbabic@denx.de>
---
 config | 60 +++++++++++++++++++++++-----------------------------------
 1 file changed, 24 insertions(+), 36 deletions(-)

diff --git a/config b/config
index 63c980a..bf07e73 100644
--- a/config
+++ b/config
@@ -27,7 +27,7 @@ DISPLAY_DRIVER ?= x11
 # Uncomment TSYS_POSIX if your platform is not listed in include/tek/config.h
 # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 
-CC = $(CROSS_COMPILE)gcc -fpic # -DTSYS_POSIX
+CC ?= $(CROSS_COMPILE)gcc -fpic # -DTSYS_POSIX
 
 # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 # For Windows using MinGW:
@@ -127,29 +127,34 @@ TEKLIB_DEFS = -DENABLE_LAZY_SINGLETON
 #------------------------------------------------------------------------------
 
 PREFIX ?= /usr/local
-LUAVER ?= 5.1
+LUAVER ?= `pkg-config --modversion lua | cut -d'.' -f1,2`
 
 LUA_DIR = $(PREFIX)
 LUA_INCDIR = $(LUA_DIR)/include
-LUA_DEFS = -I$(LUA_INCDIR)
+#LUA_DEFS = -I$(LUA_INCDIR)
 # other known paths - better comment out what is inappropriate:
-LUA_DEFS += -I/usr/include/lua$(LUAVER)
-LUA_DEFS += -I/usr/local/include/lua51
+#LUA_DEFS += -I/usr/include/lua$(LUAVER)
+#LUA_DEFS += -I/usr/local/include/lua51
+LUA_DEFS = `pkg-config --cflags lua` #$(LUA_DIR)/include
 
-LUA_LIB = $(PREFIX)/lib/lua/$(LUAVER)
+LUA_LIB ?= $(PREFIX)$(BASELIB)/lua/$(LUAVER)
 LUA_SHARE = $(PREFIX)/share/lua/$(LUAVER)
-SYS_LUA_LIB = $(PREFIX)/lib/lua/$(LUAVER)
-SYS_LUA_SHARE = $(PREFIX)/share/lua/$(LUAVER)
+SYS_LUA_LIB = $(SYS_LUA_LIB)
+SYS_LUA_SHARE = $(LUA_SHARE)
 
 #------------------------------------------------------------------------------
 # Libraries:
 #------------------------------------------------------------------------------
 
-FREETYPE_DIR = $(PREFIX)
-FREETYPE_INCDIR = $(FREETYPE_DIR)/include
-FREETYPE_LIBDIR = $(FREETYPE_DIR)/lib
-FREETYPE_LIBS = -L$(FREETYPE_LIBDIR) -lfreetype # $(CC_LD_RT)$(FREETYPE_LIBDIR)
-FREETYPE_DEFS = -I$(FREETYPE_INCDIR)/freetype2 `freetype-config --cflags`
+#FREETYPE_DIR = $(PREFIX)
+#FREETYPE_INCDIR = $(FREETYPE_DIR)/include
+#FREETYPE_LIBDIR = $(FREETYPE_DIR)/lib
+#FREETYPE_LIBS = -L$(FREETYPE_LIBDIR) -lfreetype # $(CC_LD_RT)$(FREETYPE_LIBDIR)
+#FREETYPE_DEFS = -I$(FREETYPE_INCDIR)/freetype2 `freetype-config --cflags`
+
+FREETYPE_INCDIR = `pkg-config --cflags freetype2`
+FREETYPE_LIBS = `pkg-config --libs freetype2`
+FREETYPE_DEFS = $(FREETYPE_INCDIR)
 
 # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 # X11 defines and libraries:
@@ -174,7 +179,8 @@ XFT_DEFS = -I$(XFT_INCDIR) -DENABLE_XFT
 FONTCONFIG_DIR = $(PREFIX)
 FONTCONFIG_INCDIR = $(FONTCONFIG_DIR)/include
 FONTCONFIG_LIBDIR = $(FONTCONFIG_DIR)/lib
-FONTCONFIG_LIBS = -L$(FONTCONFIG_LIBDIR) -lfontconfig # $(CC_LD_RT)$(FONTCONFIG_LIBDIR)
+#FONTCONFIG_LIBS = -L$(FONTCONFIG_LIBDIR) -lfontconfig # $(CC_LD_RT)$(FONTCONFIG_LIBDIR)
+FONTCONFIG_LIBS = -L$(FONTCONFIG_LIBDIR) # -lfontconfig $(CC_LD_RT)$(FONTCONFIG_LIBDIR)
 FONTCONFIG_DEFS = -I$(FONTCONFIG_INCDIR)
 
 # - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
@@ -237,7 +243,7 @@ MODDIR = $(BINDIR)/mod
 #------------------------------------------------------------------------------
 
 DEBUG = -DTDEBUG=5 -g # TDEBUG: the lower, the more verbose. 5 = WARN
-WARN = -Wall -Wextra -Wno-unused-parameter -Wno-unused-function -Wno-long-long -Wno-unused-value
+WARN = -Wall -Wextra -Wno-unused-parameter -Wno-unused-function -Wno-long-long # -Wno-unused-value
 OPT = -O2 -fno-strict-aliasing
 INCL = -I. -Iinclude -I$(INCDIR)
 LIBCFLAGS = $(DEBUG) $(WARN) $(OPT) $(INCL) $(EXTRADEFS) $(CPPFLAGS) $(CFLAGS)
@@ -250,7 +256,8 @@ BINCFLAGS = $(DEBUG) $(WARN) $(OPT) $(INCL) $(EXTRADEFS) $(LDFLAGS)
 # Build tools:
 #------------------------------------------------------------------------------
 
-AR = $(CROSS_COMPILE)ar rcu
+#AR = $(CROSS_COMPILE)ar rcu
+AR = ar rcu
 RM = rm -f
 RMDIR = rm -rf
 MKDIR = mkdir -p
@@ -258,27 +265,8 @@ ECHO = echo
 INSTALL_B = install
 INSTALL_F = install
 INSTALL_D = install -d
-INSTALL_S = install -s
-STRIP = $(CROSS_COMPILE)strip
+INSTALL_S = install
 
-#------------------------------------------------------------------------------
-# Lua/tekUI standalone executable: not needed by default, only for 'make tools'
-#------------------------------------------------------------------------------
-
-# Lua distribution path:
-LUADISTDIR = $(BASEDIR)/../lua-5.1.5
-# Linux:
-LUAEXE_DEFS = $(X11_DEFS) $(TEKUI_DEFS) -DLUA_USE_LINUX
-LUAEXE_LIBS = $(X11_LIBS) -lreadline -lm -ldl -pthread
-# Mac OS X:
-# LUAEXE_DEFS = $(X11_DEFS) $(TEKUI_DEFS) -DLUA_USE_MACOSX
-# LUAEXE_LIBS = $(X11_LIBS) -lreadline
-# Windows:
-# LUAEXE_DEFS = $(TEKUI_DEFS)
-# LUAEXE_LIBS = $(WIN_LIBS)
-# FreeBSD:
-# LUAEXE_DEFS = $(X11_DEFS) $(TEKUI_DEFS) -DLUA_USE_LINUX
-# LUAEXE_LIBS = -Wl,-E $(X11_LIBS) -lreadline -lm -pthread
 
 #------------------------------------------------------------------------------
 # Predefined targets:
-- 
2.25.1

