From e6850fcf6273b2068c5ae659bbacd8c7f664fd26 Mon Sep 17 00:00:00 2001
From: Stefano Babic <sbabic@denx.de>
Date: Mon, 22 Jun 2015 14:58:24 +0200
Subject: [PATCH] Allow fw env tools to be available as library

Signed-off-by: Stefano Babic <sbabic@denx.de>
---
 tools/env/Makefile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/tools/env/Makefile b/tools/env/Makefile
index 40164f7..38ad118 100644
--- a/tools/env/Makefile
+++ b/tools/env/Makefile
@@ -23,10 +23,12 @@ endif
 always := fw_printenv
 hostprogs-y := fw_printenv
 
-fw_printenv-objs := fw_env.o fw_env_main.o \
+lib-y += fw_env.o \
 	crc32.o ctype.o linux_string.o \
 	env_attr.o env_flags.o aes.o
 
+fw_printenv-objs := fw_env_main.o $(lib-y)
+
 quiet_cmd_crosstools_strip = STRIP   $^
       cmd_crosstools_strip = $(STRIP) $^; touch $@
 
-- 
1.9.1

