From 2e6cbd39bb99c890bcbc513dcd643c239d34f7c6 Mon Sep 17 00:00:00 2001
From: Kas User <kas@example.com>
Date: Tue, 22 May 2018 19:20:04 +0200
Subject: [PATCH] fix for OE

Signed-off-by: Kas User <kas@example.com>
---
 makefile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/makefile
+++ b/makefile
@@ -14,8 +14,10 @@ PLATS= macosx linux win32 mingw freebsd
 
 all: $(PLAT)
 
+LUAV = `pkg-config --modversion lua | cut -d'.' -f1,2`
+
 $(PLATS) none install install-unix local clean:
-	$(MAKE) -C src $@
+	$(MAKE) -C src $@ LUAV=${LUAV}
 
 print:
 	$(MAKE) -C src $@
--- a/src/makefile
+++ b/src/makefile
@@ -50,7 +50,7 @@ LDIR_macosx?=share/lua/$(LUAV)
 LUAINC_linux_base?=/usr/include
 LUAINC_linux?=$(LUAINC_linux_base)/lua/$(LUAV)
 LUAPREFIX_linux?=/usr/local
-CDIR_linux?=lib/lua/$(LUAV)
+CDIR_linux?=$(BASELIB)/lua/$(LUAV)
 LDIR_linux?=share/lua/$(LUAV)
 
 # LUAINC_freebsd:
